src/pynwb/core.py

```diff
@@ -166,8 +144,27 @@ def __gather_nwbfields(cls, name, bases, classdict):
             new_nwbfields.append(pname)
         cls.__nwbfields__ = tuple(new_nwbfields)
 
-    def __str__(self):
-        return nwb_repr(self)
+    def __repr__(self):
+        template = "{} {}\nFields:\n""".format(getattr(self, 'name'), type(self))
+        for k, v in iteritems(self.fields):
+            template += "  {}: {} \n".format(k, self.smart_str(v))
+        return template
+
+    def smart_str(self, v):
```

Comment:
Could you add a docstring for the ``smart_str`` function to describe what it does. Also should smart_str be a private or a public function, i.e., should to be renamed it __smart_str if it is only used internally